### PR TITLE
Wf

### PR DESCRIPTION
**Pull request by btalberg**
We revisited retention and added the exit reason functionality! I also realized I had continued work against this branch last year and hadn't requested a merge. Doing that now.

### CODE REVIEW

**Comment by AndrewSouthpaw** [comment]
Thanks Ben! @brianjscoles and I will take a look and get back to you within a week.

I'm assuming this isn't actively blocking you, yes?

**Comment by btalberg** [comment]
I'd be grateful for eyes on this (although the script here is pretty
rough)! No rush on your end at all, this PR isn't blocking me. Thank you
thank you Andrew!

*Benjamin Jaffe-Talberg*
Pronouns: he/him/his

File: pull_retention_rates.rb [hunk 1]
```diff
@@ -98,16 +111,38 @@ def load_children(tc, school, session)
...
 
   children = tc.get 'children.json', params: { session_id: session['id'] }
-  children.each { |child| child['school'] = school['name'] }
 
   children.each do |child|
+    child['school'] = school
+
     child['ignore'] = is_child_ignored(
-        school_id=school['id'],
+        school=school,
```
Comment (AndrewSouthpaw) [inline pull_retention_rates.rb]: This seems redundant?

File: pull_retention_rates.rb [hunk 2]
```diff
@@ -98,16 +111,38 @@ def load_children(tc, school, session)
...
         classroom_ids=child['classroom_ids'],
+        child=child)
```
Comment (AndrewSouthpaw) [inline pull_retention_rates.rb]: And here?

**Review by AndrewSouthpaw — APPROVED** [review]
Hey Ben! Nothing jumped out at me as being an issue, and my current understanding is that you're the primary consumer of the API so if it seems to be working for you, I call this good. 👍

**Comment by btalberg** [comment]
Thanks @AndrewSouthpaw for taking some time to review!

**Pull request merged**